agent/consul/leader_peering.go

```diff
@@ -238,8 +245,15 @@ func (s *Server) establishStream(ctx context.Context, logger hclog.Logger, peer
 
 		logger.Trace("dialing peer", "addr", addr)
 		conn, err := grpc.DialContext(retryCtx, addr,
-			grpc.WithBlock(),
+			// TODO(peering): use a grpc.WithStatsHandler here?)
 			tlsOption,
+			// For keep alive parameters there is a larger comment in ClientConnPool.dial about that.
+			grpc.WithKeepaliveParams(keepalive.ClientParameters{
+				Time:    30 * time.Second,
+				Timeout: 10 * time.Second,
+				// send keepalive pings even if there is no active streams
+				PermitWithoutStream: true,
+			}),
```

Comment:
@rboyer do you need to set the enforcement policy on the external gRPC server (https://github.com/hashicorp/consul/blob/main/agent/grpc-external/server.go#L15) similar to how it is set on the internal server (https://github.com/hashicorp/consul/blob/main/agent/grpc-internal/handler.go#L37-L39)?

Reply:
I've confirmed that we'll need to set this. Otherwise we get an error after a while and it closes the connection:

```
INFO[0108] [transport] transport: Got too many pings from the client, closing the connection.  system=system
```

Reply:
Fixed by https://github.com/hashicorp/consul/pull/13796